src/hyperdeck/response-handler.js

```diff
@@ -12,73 +12,78 @@ var ASYNC_RESPONSE_REGEX = /^(5[0-9]{2} )/;
 function ResponseHandler(clientSocket) {
   var destroyed = false;
   var notifier = new events.EventEmitter();
-  var buffer = "";
-  var nextResponseIsMultiline = false;
-  
-  function isBufferComplete() {
-    var lines = buffer.split("\r\n");
-    // there will always be an empty element at the end as every line will always end in "\r\n"
-    // so remove it.
-    // i.e. "1\r\n2\r\n" => ["1", "2", ""]
-    lines.pop();
-    if (lines.length === 1) {
-      var asyncResponse = ASYNC_RESPONSE_REGEX.exec(lines[0]);
-      if (!asyncResponse) {
-        if (nextResponseIsMultiline) {
-          logger.debug('Assuming buffer is not complete because nextResponseIsMultiline flag is set.');
-          nextResponseIsMultiline = false;
-          return false;
-        }
-      }
-      // is it a single line response?
-      return lines[0].indexOf(":") !== lines[0].length-1;
-    } 
-    // multi line response so waiting for a blank line to signify end
-    return lines[lines.length-1] === "";
+  var buffer = [];
+
+  function isRespComplete() {
+    let complete = false;
+
+    if (buffer.length === 1) {
+      // a single line response
+      // single line response first line should never ends with ":" and
+      // 1XX and 200 is always single line response
+      complete = buffer[0].search(/^1\d\d\s|200\sok$/) === 0 && buffer[0].endsWith(':') === false;
+    } else if (buffer.length >= 3) {
+      // multi line response, so waiting for a blank line to signify end
+      complete = buffer[buffer.length - 1] === '';
```

Comment:
I think it is nicer just to `return buffer[buffer.length - 1] === ''` here, and `return false` below

Reply:
:)

I don't like multiple returns in a function.

Reply:
I agree if the function is big, but in this case it would only be a few lines and by having 3 returns it means we don't need the `complete` variable. If you want to keep it as a single return though I'm happy

Reply:
function can grow later even if its 3 loc now, if you change when it grows then changes will be many. 
In general if a function returns something then I always assign a variable if logic needs to update we only need to update that variable.

Reply:
Fair enough. I see both sides. I don't think it will get longer, and if it did it's easy to switch then ¯\_(ツ)_/¯ either way it's a really minor thing

Reply:
it about habit :)